composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/ui/CreateEventScreen.kt

```diff
@@ -256,4 +288,92 @@ fun CreateEventScreen(
             TimePicker(state = timePickerState)
         }
     }
+
+    if (viewState.showAttendeeDialog) {
+        TaskyAttendeeDialog(
+            onDismiss = {
+                onUserAction(CreateEventUserAction.DismissAttendeeDialog)
+            },
+            onConfirm = {
+                onUserAction(CreateEventUserAction.ConfirmAttendeeEmail(it))
+            },
+            onTextChanged = {
+                onUserAction(CreateEventUserAction.AttendeeEmailChange(it))
+            },
+            isError = viewState.isAttendeeEmailError
+        )
+    }
+}
+
+@Composable
+private fun TaskyAttendeeDialog(
+    onDismiss: () -> Unit,
+    onConfirm: (String) -> Unit,
+    onTextChanged: (String) -> Unit,
+    isError: Boolean
+) {
+    var text by rememberSaveable { mutableStateOf("") }
```

Comment:
I guess it makes sense for the Composable component to handle the state of text. Any thoughts?

Reply:
Not a fan of this, since
1. You lose flexibility. There's no way for you to programatically change the text field's content. Not necessary for this attendee dialog, but in general wanting to do that is not super rare.
2. You can't test this logic anymore without having to use UI tests. Imagine, you'd want to test that your ViewModel properly validates the email. Having this state in a ViewModel, you could just set it to a valid email and check if the isValid state is properly updated. Having it in the UI forces you to test this together with the screen.

Reply:
@philipplackner But in this particular case when we move the handling of the input state to ViewModel, we get ourselves an overhead to take care of this text state every time we finish some actions on the dialog. For example, after we click the "ADD" button on the dialog we need to clear text that is saved in our ViewState since it outlives the lifecycle of TaskyAttendeeDialog in this case. Wouldn't it be better for such cases to keep it within the Composable, get automatic handling of disposing of the state, and not expose our codebase for potential bugs if not handled properly?